src/AggregateFunctions/IAggregateFunction.h

```diff
@@ -267,14 +267,15 @@ class IAggregateFunctionHelper : public IAggregateFunction
             const auto & flags = assert_cast<const ColumnUInt8 &>(*columns[if_argument_pos]).getData();
             for (size_t i = 0; i < batch_size; ++i)
             {
-                if (flags[i])
+                if (flags[i] && places[i])
                     static_cast<const Derived *>(this)->add(places[i] + place_offset, columns, i, arena);
             }
         }
         else
         {
             for (size_t i = 0; i < batch_size; ++i)
-                static_cast<const Derived *>(this)->add(places[i] + place_offset, columns, i, arena);
+                if (places[i])
```

Comment:
So, we can put nullptr instead of place? But wouldn't it hurt performance?

Reply:
Well, maybe.
I suppose it is a rare case, and branch predictor will help.
Let's see what tests show.